### PR TITLE
fixed an issue in download command

### PR DESCRIPTION
**Pull request by merit-maita**
## Status
- [ ] In Progress
- [x] Ready
- [ ] In Hold - (Reason for hold)

## Related Issues
fixes: https://jira-hq.paloaltonetworks.local/browse/XSUP-16035

## Description
Fixed an issue where some playbooks were downloaded as test playbooks.

## Screenshots
Paste here any images that will help the reviewer

## Must have
- [ ] Tests
- [ ] Documentation
- [ ] [VS Code Extension](https://docs.gitlab.com/ee/ci/yaml/#tags)
  - [ ] Functionality implemented in the extension - <u>\<link-to-pr-here\></u>
  - [ ] N/A - Functionality cannot be implemented in the extension because <u>\<add-reason-here\></u>

### CODE REVIEW

File: demisto_sdk/commands/download/downloader.py [hunk 1]
```diff
@@ -603,8 +603,9 @@ def file_type_to_entity(file_data: dict, file_type: str) -> str:
         :return: The file entity, for example: Integrations
         """
         if file_type and file_type == 'playbook':
-            name: str = get_entity_name_by_entity_type(file_data, PLAYBOOKS_DIR)
-            if name and 'test' in name.lower():
+            # name: str = get_entity_name_by_entity_type(file_data, PLAYBOOKS_DIR)
+            if not file_data.get('tests'):
```
Comment (dorschw) [inline demisto_sdk/commands/download/downloader.py:607]:
```suggestion
            if 'tests' not in file_data:
```
wdyt?

Comment (merit-maita) [reply demisto_sdk/commands/download/downloader.py:607]: yeah looks better

File: demisto_sdk/commands/download/downloader.py [hunk 2]
```diff
@@ -604,7 +604,8 @@ def file_type_to_entity(file_data: dict, file_type: str) -> str:
         """
         if file_type and file_type == 'playbook':
             name: str = get_entity_name_by_entity_type(file_data, PLAYBOOKS_DIR)
-            if name and 'test' in name.lower():
+            if re.match(r'\btest\b', name.lower()) or name.endswith(("Test", "_test", "_Test")) \
```
Comment (dorschw) [inline demisto_sdk/commands/download/downloader.py:607]: please have all the options (incl. `-test`, which seems to be missing) tested

Comment (dorschw) [inline demisto_sdk/commands/download/downloader.py:607]: the `\btest\b` isn't documented under the command's `README`, is it redundant or just missing from the docs? Please make sure they match.

File: demisto_sdk/commands/download/tests/downloader_test.py [hunk 3]
```diff
@@ -203,6 +203,11 @@ def test_verify_code_lang(self, code_lang, file_type, file_name, err_msg, output
 
     @pytest.mark.parametrize('data, file_type, entity', [
         ({'name': 'test-pb'}, 'playbook', TEST_PLAYBOOKS_DIR),
+        ({'name': 'playbook_testing'}, 'playbook', PLAYBOOKS_DIR),
+        ({'name': 'playbook_test'}, 'playbook', TEST_PLAYBOOKS_DIR),
+        ({'name': 'playbookTest'}, 'playbook', TEST_PLAYBOOKS_DIR),
+        ({'name': 'Testplaybook'}, 'playbook', TEST_PLAYBOOKS_DIR),
+        ({'name': 'Test-playbook'}, 'playbook', TEST_PLAYBOOKS_DIR),
```
Comment (dorschw) [inline demisto_sdk/commands/download/tests/downloader_test.py:210]: Please add cases for 

`playbook_Test`
`playbook-test`
`playbook-Test`
`Test123`
`Test_123`
`test_123`
`test-123`

File: demisto_sdk/commands/download/README.md [hunk 4]
```diff
@@ -85,7 +85,13 @@ Integrations, Scripts and Playbooks directories that does not contain a yml file
...
 For clarity, the given pack should be consistent with Content hierarchy structure with no rouge files present.
 
-We assume that test playbooks contain the `test` word in their name.
+We assume that test playbooks contain the `test` word in their name, in the following formats:
+- TestPlaybookName
+- PlaybookNameTest
+- test-playbook_name
+- test_playbook_name
+- playbook_name-test
+- playbook_name_test
```
Comment (dorschw) [inline demisto_sdk/commands/download/README.md:94]:
```suggestion
The SDK assumes the following playbooks as type TestPlaybook:
- Playbooks whose name starts with either`Test`, `Test_`, `test_`, `Test-`, or `test-` 
- Playbooks whose name ends with either `Test`,`_test` or `-test`.

```

File: demisto_sdk/commands/download/README.md [hunk 5]
```diff
@@ -104,6 +110,7 @@ We assume that test playbooks contain the `test` word in their name.
 
 ### Not Supported
 Integrations / Scripts written in JavaScript.
+A playbook that starts with the word 'Test', it would be downloaded as a test playbook.
```
Comment (dorschw) [inline demisto_sdk/commands/download/README.md:113]: we have 
```python
                    or name.startswith(("Test", "Test_", "test_", "Test-", "test-")):
```
is this documentation correct?

Comment (merit-maita) [reply demisto_sdk/commands/download/README.md:110]: yeah, any playbook starting with the word test in any form is considered as tpb

File: demisto_sdk/commands/download/downloader.py [hunk 6]
```diff
@@ -604,7 +604,8 @@ def file_type_to_entity(file_data: dict, file_type: str) -> str:
         """
         if file_type and file_type == 'playbook':
             name: str = get_entity_name_by_entity_type(file_data, PLAYBOOKS_DIR)
-            if name and 'test' in name.lower():
+            if name.endswith(("Test", "_test", "_Test", "-test", "-Test")) \
+                    or name.startswith(("Test", "Test_", "test_", "Test-", "test-")):
```
Comment (dorschw) [inline demisto_sdk/commands/download/downloader.py:608]:
```suggestion
                    or name.lower().startswith('test'):
```

**Review by dorschw — APPROVED** [review]
nice

**Pull request merged**